src/swap/components/SwapProvider.tsx

```diff
@@ -52,6 +57,17 @@ export function SwapProvider({
   const { useAggregator } = experimental;
   // Core Hooks
   const accountConfig = useConfig();
+  const [callsId, setCallsId] = useState<Hex>();
+  const callsStatus = useCallsStatus({
+    id: callsId || '0x',
+    query: {
+      refetchInterval: (query) => {
+        return query.state.data?.status === 'CONFIRMED' ? false : 1000;
+      },
+      enabled: callsId !== undefined,
+    },
+  });
```

Comment:
I feel like we need to abstract this. Have this kind of logic in the Provider makes it a bit harder to understand.